core/common/src/main/java/alluxio/network/TieredIdentityFactory.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.PropertyKey.Template;
+import alluxio.util.ShellUtils;
+import alluxio.wire.TieredIdentity;
+import alluxio.wire.TieredIdentity.LocalityTier;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class for getting tiered identity.
+ */
+public final class TieredIdentityFactory {
+  private static volatile TieredIdentity sInstance = null;
+  // Synchronize on this lock to modify sInstance.
+  private static final Object LOCK = new Object();
```

Comment:
Could you reorder this declaration to be first?

Reply:
done